### DISCUSSION TITLE
security issue about error component.

### DISCUSSION THREAD

**Discussion by jcyuan** [vercel/next.js#71204] [Help] [answered]
### Summary

Guys, i have a simple question: according to the Next.js documentation, error.tsx must be a client component for showing errors that you didn't catch. In that case, if a server-side logic error occurs, such as a database error, wouldn't it expose the error information to the client? Or do I misunderstand something?

![image](https://github.com/user-attachments/assets/e9fa50e2-38da-4cb0-bcbf-86a80e35f9d2)
the error was serialized and passed to client from server i believe, am I right?

**Comment by taylor-wood-morpheus** [comment, marked as answer]
See: https://nextjs.org/docs/app/building-your-application/routing/error-handling#securing-sensitive-error-information

> During production, the Error object forwarded to the client only includes a generic message and digest property.
> This is a security precaution to avoid leaking potentially sensitive details included in the error to the client.
> The message property contains a generic message about the error and the digest property contains an automatically generated hash of the error that can be used to match the corresponding error in server-side logs.

**Reply by jcyuan** [reply]
yes, i found this after post this, sorry. and thank you.
